### PR TITLE
[CELEBORN-224][FOLLOWUP] Correct license and notices.

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?

Correct licenses and notices according to vote discussion.

### Why are the changes needed?

The LICENSE and NOTICE should be updated to meet ASF policy.
https://lists.apache.org/thread/k74hffv0n737h1l379nqtb1m3mr4x9rd

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Not necessary, it does not touch the code.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @carp84 please review this pr, thanks!

**Review by carp84 — APPROVED** [review]
LGTM. Please wait for a while for other mentors' review before merging it.

**Review by pan3793 — APPROVED** [review]
LGTM, and I also updated the PR description

File: NOTICE-binary
Comment (waitinfuture) [reply NOTICE-binary:66]: > It's strange that we put this file in the binary release kit instead of the source release kit.

Hi @WillemJiang , this part is copied from the NOTICE of log4j-core-2.17.2.jar
```
Apache Log4j Core
Copyright 1999-2012 Apache Software Foundation

This product includes software developed at
The Apache Software Foundation (http://www.apache.org/).

ResolverUtil.java
Copyright 2005-2006 Tim Fennell
```
And since Celeborn's code doesn't depend on the class, we think it should be in the binary release kit instead of the source release kit.

Please correct me if I misunderstood, thanks!

Comment (WillemJiang) [reply NOTICE-binary:66]: OK, thanks for the explanation.

**Pull request merged**
